[Bug]: Nil Pointer Reference on aws_quicksight_account_subscription

### Terraform Core Version

1.6.2

### AWS Provider Version

5.60.0

### Affected Resource(s)

aws_quicksight_account_subscription

### Expected Behavior

The aws_quicksight_account_subscription resource should show applied successfully.

### Actual Behavior

The Terraform process completes with an error.

For my use case the QuickSight subscription already exists, so I do not know if the QuickSight subscription would be created if it did not exist already.

There can only be one QuickSight subscription per AWS account so I cannot test with no existing QuickSight subscription without deleting the one I have (which contains data and dashboards).

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.60.0"
    }
  }
  required_version = ">= 1.6.2"
}

provider "aws" {
}

resource "aws_quicksight_account_subscription" "default" {
  account_name          = "my-account-name"
  admin_group           = [
    "SSO-Admin-Group-Name",
  ]
  authentication_method = "IAM_IDENTITY_CENTER"
  author_group          = [
    "SSO-Author-Group-Name",
  ]
  edition               = "ENTERPRISE"
  notification_email    = "someone@acme.com"
  reader_group          = [
    "SSO-Reader-Group-Name",
  ]
}
```

### Steps to Reproduce

Terraform apply on the above. Will be hard to repro given QuickSight dependencies on SSO (IAM Identity Centre) setup.

### Debug Output

```
Initializing the backend...

Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Reusing previous version of hashicorp/aws from the dependency lock file
- Installing hashicorp/aws v5.60.0...
- Installed hashicorp/aws v5.60.0 (signed by HashiCorp)

Terraform has made some changes to the provider dependency selections recorded
in the .terraform.lock.hcl file. Review those changes and commit them to your
version control system if they represent changes you intended to make.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
aws_quicksight_account_subscription.default: Refreshing state... [id=249663324982]

Planning failed. Terraform encountered an error while generating this plan.
╷
│ Error: Plugin did not respond
│ 
│   with aws_quicksight_account_subscription.default,
│   on resources-quicksight.tf line 1, in resource "aws_quicksight_account_subscription" "default":
│    1: resource "aws_quicksight_account_subscription" "default" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵


Stack trace from the terraform-provider-aws_v5.60.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x10e3a71f]

goroutine 45 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/quicksight.resourceAccountSubscriptionRead({0x16c89f48, 0xc0010d68a0}, 0xc004506b80, {0x14a8be20?, 0xc0053c05b0?})
	github.com/hashicorp/terraform-provider-aws/internal/service/quicksight/account_subscription.go:237 +0x31f
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Read.interceptedHandler[...].func9(0xc004506b80?, {0x14a8be20?, 0xc0053c05b0})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x16c89f48?, {0x16c89f48?, 0xc0042f1290?}, 0xd?, {0x14a8be20?, 0xc0053c05b0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:818 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc002097b20, {0x16c89f48, 0xc0042f1290}, 0xc003ef4340, {0x14a8be20, 0xc0053c05b0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:1117 +0x529
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0053a8300, {0x16c89f48?, 0xc0042f11d0?}, 0xc0045528c0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:708 +0x6c5
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0xc0012da380, {0x16c89f48?, 0xc0042f0f00?}, 0xc0045528c0)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ReadResource.go:35 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0012626e0, {0x16c89f48?, 0xc004551650?}, 0xc0014878f0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:783 +0x309
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x14878c00, 0xc0012626e0}, {0x16c89f48, 0xc004551650}, 0xc004506480, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:482 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0019a0000, {0x16c89f48, 0xc0045515c0}, {0x16cec2a0, 0xc00052e180}, 0xc004561200, 0xc000e15b90, 0x20272228, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc0019a0000, {0x16cec2a0, 0xc00052e180}, 0xc004561200)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 25
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.60.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.


Error: Process completed with exit code 1.
```

### Panic Output

_No response_

### Important Factoids

- This was working in a previous AWS Provider version as I used it to create my original QuickSight subscription a few months ago.
- Might be related to issue [34199](https://github.com/hashicorp/terraform-provider-aws/issues/34199).


### References

_No response_

### Would you like to implement a fix?

No

--- a/.changelog/38752.txt
+++ b/.changelog/38752.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_quicksight_account_subscription: Fix panic when read returns nil account info
+```

--- a/internal/service/quicksight/account_subscription.go
+++ b/internal/service/quicksight/account_subscription.go
@@ -228,12 +228,13 @@ func resourceAccountSubscriptionRead(ctx context.Context, d *schema.ResourceData
 
 	out, err := FindAccountSubscriptionByID(ctx, conn, d.Id())
 
-	if !d.IsNewResource() && tfresource.NotFound(err) {
+	var ere *tfresource.EmptyResultError
+	if !d.IsNewResource() && (tfresource.NotFound(err) || errors.As(err, &ere)) {
 		log.Printf("[WARN] QuickSight AccountSubscription (%s) not found, removing from state", d.Id())
 		d.SetId("")
 		return diags
 	}
-	// Ressource is logically deleted with UNSUBSCRIBED status
+	// Resource is logically deleted with UNSUBSCRIBED status
 	if !d.IsNewResource() && aws.StringValue(out.AccountSubscriptionStatus) == statusUnsuscribed {
 		log.Printf("[WARN] QuickSight AccountSubscription (%s) unsuscribed, removing from state", d.Id())
 		d.SetId("")
@@ -352,7 +353,7 @@ func FindAccountSubscriptionByID(ctx context.Context, conn *quicksight.QuickSigh
 		return nil, err
 	}
 
-	if out == nil || out.AccountInfo.AccountName == nil {
+	if out == nil || out.AccountInfo == nil {
 		return nil, tfresource.NewEmptyResultError(in)
 	}
 

